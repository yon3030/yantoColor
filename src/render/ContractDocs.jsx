import React from 'react';

import { FunctionDocs } from './FunctionDocs';
import { FunctionIdentifier } from './FunctionIdentifier';

export function ContractDocs(props) {
  const { name, devdoc, functions, events } = props;

  const functionDocs = functions.map(fn => (
    <FunctionDocs key={ fn.identifier } {...fn} contractName={ name } />
  ));

  const functionIndex = functions.length > 0 && <>
    <span className="contract-index-title">Functions</span>
    <ul>
      {
        functions.map(fn => (
          <li key={ fn.identifier }>
            <a href={ '#' + name + '.' + fn.identifier }>
              <FunctionIdentifier { ...fn } />
            </a>
          </li>
        ))
      }
    </ul>
  </>;

  const eventDocs = events.map(fn => (
    <FunctionDocs key={ fn.identifier } {...fn} contractName={ name } />
  ));

  const eventIndex = events.length > 0 && <>
    <span className="contract-index-title">Events</span>
    <ul>
      { 
        events.map(fn => (
          <li key={ fn.identifier }>
            <a href={ '#' + name + '.' + fn.identifier }>
              <FunctionIdentifier { ...fn } />
            </a>
          </li>
        ))
      }
    </ul>
  </>;

  return (
    <>
    { '\n\n' }
    ### `{ name }`
    { '\n\n' }
    { devdoc }
    { '\n\n' }
    <div className="contract-index">
      { functionIndex }
      { eventIndex }
    </div>
    { '\n\n' }
    { functionDocs }
    { '\n\n' }
    { eventDocs }
    { '\n\n' }
    </>
  );
}
