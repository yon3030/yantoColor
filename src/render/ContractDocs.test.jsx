import React from 'react';
import renderer from 'react-test-renderer';

import { ContractDocs } from './ContractDocs';

test('no functions', function () {
  const contract = {
    name: 'Foo',
    devdoc: 'devdoc',
    functions: [],
  };

  const tree = renderer
    .create(<ContractDocs { ...contract } />)
    .toJSON();

  expect(tree).toMatchSnapshot();
});

test('1 function', function () {
  const foo = {
    identifier: 'foo()',
    astNode: {
      nodeType: 'FunctionDefinition',
      name: 'foo',
      parameters: [],
    },
    devdoc: 'foo.devdoc',
  };

  const contract = {
    name: 'Foo',
    devdoc: 'devdoc',
    functions: [ foo ],
  };

  const tree = renderer
    .create(<ContractDocs { ...contract } />)
    .toJSON();

  expect(tree).toMatchSnapshot();
});
