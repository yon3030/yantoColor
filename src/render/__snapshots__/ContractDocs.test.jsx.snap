// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`1 function 1`] = `
Array [
  <h2>
    Foo
  </h2>,
  "devdoc",
  <h2>
    foo()
  </h2>,
  "foo.devdoc",
]
`;

exports[`no functions 1`] = `
Array [
  <h2>
    Foo
  </h2>,
  "devdoc",
]
`;
